controllers/cassandrarestore_controller.go

```diff
@@ -162,16 +162,48 @@ func (r *CassandraRestoreReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 			}
 		}
 
-		if err = setBackupNameInRestoreContainer(backup.Spec.Name, cassdc); err != nil {
+		var podTemplateSpecUpdated bool
```

Comment:
I just learned that Go does not support bitwise operations for bool, but It should be equivalent to bitwise-OR. It really does not matter if one is true and the other is false or vice-versa. The important thing is whether `PodTemplateSpec` is updated. Maybe it would be better to compare the hash of `PodTemplateSpec` before and after those calls.